src/algebra/jordan/special.lean

```diff
@@ -0,0 +1,185 @@
+
+
+instance [has_neg α] : has_neg αˢʸᵐ :=
+{ neg := λ a, sym ( -(unsym a)) }
+
+instance {R : Type*} [has_scalar R α] : has_scalar R αˢʸᵐ :=
+{ smul := λ r a, sym (r•(unsym a)) }
+
+@[simp] lemma sym_smul {R : Type*} [has_scalar R α] (c : R) (a : α) : sym (c • a) = c • sym a := rfl
+
+lemma unsym_injective : function.injective (@unsym α) := λ _ _, id
+
+instance [add_comm_semigroup α] : add_comm_semigroup (αˢʸᵐ) :=
+unsym_injective.add_comm_semigroup _ (λ _ _, rfl)
+
+instance [add_monoid α] : add_monoid (αˢʸᵐ) :=
+unsym_injective.add_monoid_smul _ rfl (λ _ _, rfl) (λ _ _, rfl)
+
+instance [add_group α] : add_group (αˢʸᵐ) :=
+unsym_injective.add_group_smul _ rfl
+  (λ _ _, rfl) (λ _, rfl) (λ _ _, rfl) (λ _ _, rfl) (λ _ _, rfl)
+
+instance [add_comm_monoid α] : add_comm_monoid (αˢʸᵐ) :=
+{ ..sym_alg.add_comm_semigroup, ..sym_alg.add_monoid }
+
+instance [add_comm_group α] : add_comm_group (αˢʸᵐ) :=
+{ ..sym_alg.add_comm_monoid, ..sym_alg.add_group }
+
+@[simp] lemma sym_zero [has_zero α] : sym (0 : α) = 0 := rfl
+@[simp] lemma unsym_zero [has_zero α] : unsym (0 : αˢʸᵐ) = 0 := rfl
+
+@[simp] lemma unsym_eq_zero_iff {α} [has_zero α] (a : αˢʸᵐ) : a.unsym = (0 : α) ↔ a = (0 : αˢʸᵐ) :=
+unsym_injective.eq_iff' rfl
+
+lemma zero_zero [ring α] : unsym add_comm_group.zero = (0:α) := rfl
+
+instance {R : Type*} [semiring R] [add_comm_monoid α] [module R α] : module R αˢʸᵐ :=
+function.injective.module R ⟨unsym, rfl, λ _ _, rfl⟩ (λ _ _, id) (λ _ _, rfl)
+
+/- Introduce the symmetrised multiplication-/
+noncomputable instance [ring α] [algebra ℝ α] : has_mul(αˢʸᵐ) :=
+{ mul := λ a b, (sym ((1/2:ℝ)•(unsym(a)*unsym(b)+unsym(b)*unsym(a)))), }
+
+lemma mul_def [ring α] [algebra ℝ α] (a b: αˢʸᵐ) :
+  a*b = sym ((1/2:ℝ)•(unsym(a)*unsym(b)+unsym(b)*unsym(a))) := by refl
+
+/- The symmetrisation of a real (unital, associative) algebra is a non-associative ring -/
+noncomputable instance [ring α] [algebra ℝ α] : non_unital_non_assoc_ring (αˢʸᵐ) :=
+{ zero_mul := λ _,
+  begin
+    simp only [mul_def,zero_zero, add_zero, sym_zero, zero_mul, mul_zero, smul_zero],
+    exact rfl,
+  end,
+  mul_zero :=  λ _,
+  begin
+    simp only [mul_def,zero_zero, add_zero, sym_zero, zero_mul, mul_zero, smul_zero],
+    exact rfl,
+  end,
+  left_distrib := λ a b c, begin
+    change (1/2:ℝ)•(unsym(a)*(unsym(b)+unsym(c))+(unsym(b)+unsym(c))*unsym(a)) =
+      (1/2:ℝ)•(unsym(a)*unsym(b)+unsym(b)*unsym(a))+(1/2:ℝ)•(unsym(a)*unsym(c)+unsym(c)*unsym(a)),
+    rw [mul_add, add_mul, smul_add, smul_add, smul_add, smul_add, smul_add],
+    abel,
+  end,
+  right_distrib := λ a b c, begin
+    change (1/2:ℝ)•((unsym(a)+unsym(b))*unsym(c)+unsym(c)*(unsym(a)+unsym(b))) =
+      (1/2:ℝ)•(unsym(a)*unsym(c)+unsym(c)*unsym(a))+(1/2:ℝ)•(unsym(b)*unsym(c)+unsym(c)*unsym(b)),
+    rw [mul_add, add_mul, smul_add, smul_add, smul_add, smul_add, smul_add],
+    abel,
+  end,
+  ..sym_alg.has_mul,
+  ..sym_alg.add_comm_group, }
+
+
+theorem two_rmul [ring α] [algebra ℝ α]  (x:α) : (2 : ℝ) • x = x + x :=
```

Comment:
This already exists, it's `two_smul`

Reply:
I wasn't able to see how I'd use `two_smul` to prove `sym_squares`?

Reply:
Although switching to `[invertible (2 : α)]` removes this problem anyway :)